stm/stm.ml

```diff
@@ -1723,7 +1723,8 @@ end = struct (* {{{ *)
     | `ERROR -> exit 1
     | `ERROR_ADMITTED -> cst, false
     | `OK_ADMITTED -> cst, false
-    | `OK { Declare.name } ->
+    | `OK po ->
+        let name = Declare.get_po_name po in
```

Comment:
IMO we should get the name out before returning OK instead of after
ie instead of 
~~~ocaml
let check_task_aux ... = 
  ...
  `OK po

...

    | `OK po ->
        let name = Declare.get_po_name po in
~~~
do
~~~ocaml

let check_task_aux ... = 
  ...
  let name = Declare.get_po_name po in
  `OK name

...

    | `OK name ->
~~~

Reply:
Good point, done. In fact I wonder if this whole function is redundant as some `name` is already in scope; the STM should keep better track of this information, see the problems we have now in `save_lemma_proved_delayed` but that's left for other PR.